test/server/cards/01_SOR/units/LieutenantChildsenDeathStarPrisonWarden.spec.ts

```diff
@@ -0,0 +1,153 @@
+                    expect(context.player2).toBeActivePlayer();
+
+                    reset();
+
+                    // Reveal 3 Vigilance cards
+                    context.player1.clickCard(context.lieutenantChildsen);
+                    expect(context.player1).toBeAbleToSelectExactly([context.pykeSentinel, context.systemPatrolCraft, context.cargoJuggernaut, context.resilient, context.vanquish]);
+                    context.player1.clickCard(context.cargoJuggernaut);
+                    context.player1.clickCard(context.vanquish);
+                    context.player1.clickCard(context.resilient);
+                    context.player1.clickPrompt('Done');
+
+                    expect(context.getChatLogs(2)).toEqual([
+                        'player1 uses Lieutenant Childsen to reveal 3 cards',
+                        'player1 reveals Cargo Juggernaut, Vanquish, Resilient due to Lieutenant Childsen',
+                    ]);
+
+                    expect(context.lieutenantChildsen).toHaveExactUpgradeNames(['experience', 'experience', 'experience']);
+                    expect(context.player2).toBeActivePlayer();
+
+                    reset();
+
+                    // Reveal 4 Vigilance cards
+                    context.player1.clickCard(context.lieutenantChildsen);
+                    expect(context.player1).toBeAbleToSelectExactly([context.pykeSentinel, context.systemPatrolCraft, context.cargoJuggernaut, context.resilient, context.vanquish]);
+                    context.player1.clickCard(context.cargoJuggernaut);
+                    context.player1.clickCard(context.vanquish);
+                    context.player1.clickCard(context.resilient);
+                    context.player1.clickCard(context.pykeSentinel);
+                    context.player1.clickPrompt('Done');
+
+                    expect(context.getChatLogs(2)).toEqual([
+                        'player1 uses Lieutenant Childsen to reveal 4 cards',
+                        'player1 reveals Cargo Juggernaut, Vanquish, Resilient, Pyke Sentinel due to Lieutenant Childsen',
+                    ]);
+
+                    expect(context.lieutenantChildsen).toHaveExactUpgradeNames(['experience', 'experience', 'experience', 'experience']);
+                    expect(context.player2).toBeActivePlayer();
+
+                    reset();
+
+                    // Reveal no Vigilance cards
+                    context.player1.clickCard(context.lieutenantChildsen);
+                    expect(context.player1).toBeAbleToSelectExactly([context.pykeSentinel, context.systemPatrolCraft, context.cargoJuggernaut, context.resilient, context.vanquish]);
+                    context.player1.clickPrompt('Done');
+
+                    expect(context.lieutenantChildsen).toHaveExactUpgradeNames([]);
+                    expect(context.player2).toBeActivePlayer();
+                });
+            });
+
+            describe('when hand does not contain any Vigilance card', function() {
```

Comment:
Now that we have Childsen, there's a scenario we get asked about all the time it would be great to have a test for if you don't mind. Could you go to `DefeatTiming.spec.ts` and add a case confirming that if there is an enemy Snoke on the field, that Childsen dies despite his ability? I'm confident it will work but it comes up so much we might as well have a test we can point to.

Reply:
I didn't even know it worked like that! Good to know!